realsense2_camera/include/realsense_node_factory.h

```diff
@@ -47,7 +47,8 @@ namespace realsense2_camera
     {
         ROS_INFO_STREAM(strsignal(signum) << " Signal is received! Terminating RealSense Node...");
         ros::shutdown();
-        exit(signum);
+        sleep(5);
```

Comment:
Can you provide comments for this change which describe the situation this resolves.  Is this to prevent thrashing?

Reply:
exit(1) is not thread safe and should not be invoked in a multithreaded environment unless all the threads had been joined. So yes this prevents segmentation faults.